apistrap/decorators.py

```diff
@@ -126,40 +126,50 @@ def __call__(self, wrapped_func: Callable):
         innermost_func = _get_wrapped_function(wrapped_func)
         self.outermost_decorators[innermost_func] = self
 
-        @wraps(wrapped_func)
-        def wrapper(*args, **kwargs):
-            response = wrapped_func(*args, **kwargs)
-
-            if isinstance(response, Response):
-                return response
-            if not isinstance(response, self._response_class):
-                if self.outermost_decorators[innermost_func] == self:
-                    raise UnexpectedResponseError(type(response))
-                return response
-            if isinstance(response, FileResponse):
-                return send_file(filename_or_fp=response.filename_or_fp,
-                                 mimetype=self._mimetype,
-                                 as_attachment=response.as_attachment,
-                                 attachment_filename=response.attachment_filename,
-                                 add_etags=response.add_etags,
-                                 cache_timeout=response.cache_timeout,
-                                 conditional=response.conditional,
-                                 last_modified=response.last_modified)
-
-            try:
-                response.validate()
-            except DataError as e:
-                raise InvalidResponseError(e.errors) from e
-
-            response = jsonify(response.to_primitive())
-            response.status_code = self._code
-            return response
+        if inspect.iscoroutinefunction(wrapped_func):
+            @wraps(wrapped_func)
+            async def wrapper(*args, **kwargs):
```

Comment:
wow, one can also define coroutines inside normal functions?

Reply:
Yep, there's actually no reason why that should be a problem - after all, it's just a weird function.